src/pycita/pycita.py

```diff
@@ -0,0 +1,696 @@
+    #     """
+    #     估计合约调用所需的quota. (只在商业版可用)
+
+    #     :param contract_addr: 合约地址.
+    #     :param func_addr: 合约内的函数地址.
+    #     :param param: 编码后的函数参数列表.
+    #     :param from_addr: 合约的调用方地址.
+    #     :return: 此调用所需的quota.
+    #     """
+    #     to_ = param_to_str(contract_addr)
+    #     assert len(to_) == 40 + 2
+    #     req = {'to': to_}
+
+    #     if from_addr:
+    #         from_ = param_to_str(from_addr)
+    #         assert len(from_) == 40 + 2
+    #         req['from'] = from_
+
+    #     data = param_to_str(func_addr)
+    #     assert len(data) == 8 + 2
+    #     if param:
+    #         data += param_to_str(param)[2:]
+    #     req['data'] = data
+
+    #     r = self._jsonrpc('estimateQuota', [req, 'latest'])
+    #     assert isinstance(r, str) and r.startswith('0x')
+    #     return ast.literal_eval(r)
+
+    def get_code(self, contract_addr: PARAM) -> bytes:
+        """
+        获取合约代码.
+
+        :param contract_addr: 合约地址, 20字节
+        :return: 合约代码bytes
+        """
+        addr = param_to_str(contract_addr)
+        assert len(addr) == 42
+        r = self._jsonrpc('getCode', [addr, 'latest'])
+        assert isinstance(r, str) and r.startswith('0x')
+        if r == '0x':  # 合约不存在
+            return b''
+        rb = param_to_bytes(r)
+        return rb
+
+    def get_abi(self, contract_addr: PARAM) -> List:
+        """
+        获取合约的ABI.
+
+        :param contract_addr: 合约地址, 20字节
+        :return: ABI的json
+        """
+        addr = param_to_str(contract_addr)
+        assert len(addr) == 42
+        r = self._jsonrpc('getAbi', [addr, 'latest'])
+        assert isinstance(r, str) and r.startswith('0x')
+        if r == '0x':  # 合约不存在或未绑定ABI
+            return []
+
+        rb = param_to_bytes(r)
+        rbs = decode_param('string', rb)
+        return json.loads(rbs)
+
+    def store_abi(self, private_key: PARAM, contract_addr: PARAM, abi: str) -> str:
+        """
+        将ABI追加给指定的合约.
+
+        :param private_key: 私钥
+        :param contract_addr: 合约地址
+        :param abi: ABI的json的字符串形式
+        :return: 交易hash
+        """
+        abi_data = encode_param('string', abi)
+        return self.send_transaction(private_key, '0xffffffffffffffffffffffffffffffffff010001', join_param(contract_addr, abi_data))
+
+    def get_transaction(self, tx_hash: PARAM) -> Dict:
+        """
+        获取交易详情.
+
+        :param tx_hash: 交易hash, 32字节
+        :return: JSON结构的交易详情
+        """
+        h = param_to_str(tx_hash)
+        assert len(h) == 64 + 2
+        r = self._jsonrpc('getTransaction', [h])
+        if not r:
+            return {}
+        return cast(Dict, r)
+
+    def get_transaction_count(self, addr: PARAM, block_number: str = 'latest') -> int:
+        """
+        获取指定账户发起的交易数量.
+
+        :param addr: 账户地址, 20字节
+        :param block_number: 区块高度, 可以是一个'0x'开头的高度字符串, 也可以是'latest'或'pending', 默认是'latest'
+        :return: 交易数量
+        """
```

Comment:
should add a validate method for `block_number`  to prevent a chain exception, e.g.
```python
if(block_number != 'latest' or block_number != 'pending' or block_number.startswith('0x', 0, 2)):
  raise ValueError("not supported block_number!")
```